Admin - Fix for venue edit page in legacy admin

denisdimitrov opened the pull request:
The form validation was not passing because owners was a required field in the validation and it was not submitting the form.

Fixes: https://github.com/sparkletown/internal-sparkle-issues/issues/343

mike-lvov commented on src/pages/Admin/Venue/DetailsForm.tsx:
```diff
@@ -204,10 +203,10 @@ export const DetailsForm: React.FC<DetailsFormProps> = ({
+              onSubmit={onSubmit}
               editing={!!venueId}
               formError={formError}
+              {...rest}
```
You do aware of `rest` overriding all of that, right? Will it cause any problems?

denisdimitrov replied:
I'll check if I can remove it, it seemed unnecessary on first glance.

0xdevalias replied:
(for context, when it was at the start of the props, the ones set afterwards would override it, but now that it’s moved to the end, it will override all the other props)

denisdimitrov replied:
> (for context, when it was at the start of the props, the ones set afterwards would override it, but now that it’s moved to the end, it will override all the other props)

Yup, it was kinda in the middle, overriding only half of them

mike-lvov reviewed (approved):
If tested, looks good

0xdevalias reviewed (commented):
Do we know why there were no validation errors being shown before? (At least that I remember seeing), and does this fix that aspect too? (doesn’t look like it will from a quick skim of the diff)

mike-lvov commented on src/pages/Admin/Venue/DetailsForm.tsx:
```diff
@@ -204,10 +209,12 @@ export const DetailsForm: React.FC<DetailsFormProps> = ({
               isSubmitting={isSubmitting}
               register={register}
               watch={watch}
-              {...rest}
-              onSubmit={onFormSubmit}
+              onSubmit={onSubmit}
               editing={!!venueId}
               formError={formError}
+              control={control}
+              handleSubmit={handleSubmit}
+              errors={errors}
             />
```
Now that you have removed `...rest` at all, are we sure that we pass every needed prop there? Did you go one-by-one to check what `rest` object contained?

denisdimitrov replied:
I didn't check the rest object, I just checked the expected Props in the component and passed only the ones being used/necessary.

mike-lvov replied:
Is it possible that rest had some `optional` props that were used and now we are not passing them?

denisdimitrov replied:
Nope, I checked the interface of the component and every prop is being passed. If the `rest` object had any other fields that means they were not being used by the component.

denisdimitrov commented:
> Do we know why there were no validation errors being shown before? (At least that I remember seeing), and does this fix that aspect too? (doesn’t look like it will from a quick skim of the diff)

It was a change of mine, I changed owners to be no longer optional in the VenueTypes, but I also changed it in the validation which is wrong.

0xdevalias commented:
> There was no validation error because there is no owners field in that form.

That makes sense, though I guess I wonder how we can have these errors show up in a ‘generic’ field or somewhere else (eg. bugsnag?) or otherwise, rather than failing silently like it was.
